src/main/java/io/github/opencubicchunks/cubicchunks/mixin/transform/typetransformer/transformer/config/ConfigLoader.java

```
@@ -0,0 +1,671 @@
+                }
+
+                constantReplacements.put(from, to);
+            }
+        }
+        return constantReplacements;
+    }
+
+    private static Number getNumber(Object from, boolean doubleSize) {
+        String s = from.toString();
+        if (doubleSize) {
+            if (s.contains(".")) {
+                return Double.parseDouble(s);
+            } else {
+                return Long.parseLong(s);
+            }
+        } else {
+            if (s.contains(".")) {
+                return Float.parseFloat(s);
+            } else {
+                return Integer.parseInt(s);
+            }
+        }
+    }
+
+    public static MethodID loadMethodID(JsonElement method, @Nullable MappingResolver map, @Nullable MethodID.CallType defaultCallType) {
+        MethodID methodID;
+        if (method.isJsonPrimitive()) {
+            String id = method.getAsString();
+            String[] parts = id.split(" ");
+
+            MethodID.CallType callType;
+            int nameIndex;
+            int descIndex;
+            if (parts.length == 3) {
+                char callChar = parts[0].charAt(0);
+                callType = switch (callChar) {
+                    case 'v' -> MethodID.CallType.VIRTUAL;
+                    case 's' -> MethodID.CallType.STATIC;
+                    case 'i' -> MethodID.CallType.INTERFACE;
+                    case 'S' -> MethodID.CallType.SPECIAL;
+                    default -> {
+                        System.err.println("Invalid call type: " + callChar + ". Using default VIRTUAL type");
+                        yield MethodID.CallType.VIRTUAL;
+                    }
+                };
+
+                nameIndex = 1;
+                descIndex = 2;
+            } else {
+                callType = MethodID.CallType.VIRTUAL;
+                nameIndex = 0;
+                descIndex = 1;
+            }
+
+            if (defaultCallType != null) {
+                callType = defaultCallType;
```

not sure if this coming after `callType` is set is intended; if it is, naming the argument `defaultCallType` seems a bit misleading when it overrides any existing call type